TLM/TLM/Geometry/Impl/SegmentEndId.cs

```diff
@@ -1,56 +1,48 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using TrafficManager.Traffic;
-
-namespace TrafficManager.Geometry.Impl {
-	public class SegmentEndId : ISegmentEndId {
-		public virtual ushort SegmentId { get; protected set; } = 0;
-		public virtual bool StartNode { get; protected set; } = false;
-
-		public SegmentEndId(ushort segmentId, bool startNode) {
-			SegmentId = segmentId;
-			StartNode = startNode;
-		}
-
-		private SegmentEndId() {
-
-		}
-
-		public virtual bool Relocate(ushort segmentId, bool startNode) {
-			SegmentId = segmentId;
-			StartNode = startNode;
-			return true;
-		}
-
-		public override bool Equals(object other) {
-			if (other == null) {
-				return false;
-			}
-			if (!(other is ISegmentEndId)) {
-				return false;
-			}
-			return Equals((ISegmentEndId)other);
-		}
-
-		public bool Equals(ISegmentEndId otherSegEndId) {
-			if (otherSegEndId == null) {
-				return false;
-			}
-			return SegmentId == otherSegEndId.SegmentId && StartNode == otherSegEndId.StartNode;
-		}
-
-		public override int GetHashCode() {
-			int prime = 31;
-			int result = 1;
-			result = prime * result + SegmentId.GetHashCode();
-			result = prime * result + StartNode.GetHashCode();
-			return result;
-		}
-
-		public override string ToString() {
-			return $"[SegmentEndId {SegmentId} @ {StartNode}]";
-		}
-	}
-}
+﻿namespace TrafficManager.Geometry.Impl {
+    using Traffic;
+
+    public class SegmentEndId : ISegmentEndId {
+        public ushort SegmentId { get; protected set; }
+
+        public bool StartNode { get; protected set; }
+
+        public SegmentEndId(ushort segmentId, bool startNode) {
+            SegmentId = segmentId;
+            StartNode = startNode;
+        }
+
+        public bool Relocate(ushort segmentId, bool startNode) {
+            SegmentId = segmentId;
+            StartNode = startNode;
+            return true;
+        }
+
+        public override bool Equals(object other) {
+            if (other == null) {
+                return false;
+            }
+
+            return other is ISegmentEndId && Equals((ISegmentEndId)other);
```

Comment:
return other != null && ...

Reply:
If type check is not needed here, then other is already not null at this point, because line 21 checks for null. Removed the check.

Reply:
Null is checked above in line 21

Reply:
@kvakvs Why split the return on to so many lines, it would be clearer to just have one return statement in this case, it will short circuit if other == null so it's not going to affect performance.

Reply:
I can change this but this is how the code was written before and my changes to it are minor. This is how it is in `master`: https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/blob/master/TLM/TLM/Geometry/Impl/SegmentEndId.cs#L27-L42

Reply:
Seems overly verbose in master IMO

Reply:
Simplified now. Also null check is eliminated because `is` check does it implicitly.